frontend/src/concepts/pipelines/content/createRun/useRunFormData.ts

```diff
@@ -181,37 +78,57 @@ export const useUpdateRunType = (
   }, [setFunction, initialData]);
 };
 
-const getParams = (pipeline?: PipelineKF): RunParam[] | undefined =>
-  pipeline
-    ? (pipeline.parameters || []).map((p) => ({ label: p.name, value: p.value ?? '' }))
-    : undefined;
+const getPipelineInputParams = (
+  version: PipelineVersionKF | undefined,
+  pipeline?: PipelineKF | undefined,
+): RunParam[] | undefined => {
+  let parameters = version?.parameters;
+
+  if (!version) {
+    parameters = pipeline?.default_version?.parameters || pipeline?.parameters;
+  }
+
+  return (parameters || []).map((p) => ({ label: p.name, value: p.value ?? '' }));
+};
 
 const useRunFormData = (
-  initialData?: PipelineRunKF | PipelineRunJobKF,
-  lastPipeline?: PipelineKF,
-  lastVersion?: PipelineVersionKF,
+  initialData?: PipelineRunKF | PipelineRunJobKF | undefined,
+  pipeline?: PipelineKF,
+  version?: PipelineVersionKF,
 ): GenericObjectState<RunFormData> => {
   const { project } = usePipelinesAPI();
 
-  const objState = useGenericObjectState<RunFormData>({
+  const runFormDataState = useGenericObjectState<RunFormData>({
     project,
     nameDesc: {
       name: initialData?.name ? `Duplicate of ${initialData.name}` : '',
       description: initialData?.description ?? '',
     },
-    pipeline: lastPipeline ?? null,
-    version: lastVersion ?? lastPipeline?.default_version ?? null,
-    // experiment: null,
+    pipeline: pipeline ?? null,
+    version: version ?? pipeline?.default_version ?? null,
     runType: { type: RunTypeOption.ONE_TRIGGER },
-    params: getParams(lastPipeline),
+    params: getPipelineInputParams(version, pipeline),
   });
-  const setFunction = objState[1];
-  useUpdatePipeline(setFunction, initialData);
-  useUpdatePipelineRun(setFunction, initialData);
-  // useUpdateExperiment(setFunction, initialData);
-  useUpdateRunType(setFunction, initialData);
+  const [, setRunFormValue] = runFormDataState;
+
+  // Update pipeline, version, and input params when the pipeline or version changes.
+  React.useEffect(() => {
+    if (pipeline) {
+      setRunFormValue('pipeline', pipeline);
+    }
+
+    if (version) {
+      setRunFormValue('version', version);
+    }
+
+    if (version || pipeline) {
+      setRunFormValue('params', getPipelineInputParams(version, pipeline));
+    }
+  }, [pipeline, version, setRunFormValue]);
```

Comment:
This is something I kind of worrying about, if we have an auto-fetch on this page, will the form data (here are pipeline and version selections) be reset back to the initial value no matter what user changes them to?

Reply:
We don't have polling setup for the requests that provide the version and pipeline, and the other source of these 2 things is from react router params, so I don't think this should be an issue, but I could be wrong.

Reply:
Yeah I tested it, it seems fine since we don't have any auto poll for the run page.

Reply:
I added conditions for each of these setters to not update if formData already exists for those values which should guard against your concern - for possibly some future case where this might happen.